From 74319fe57d7da12fa88ec0e16c380f191d676b82 Mon Sep 17 00:00:00 2001
From: Jonathan M Hsieh <jmhsieh@apache.org>
Date: Fri, 6 Nov 2015 11:39:51 -0800
Subject: [PATCH 278/295] HBASE-14780 Integration Test that run with chaos
 monkey must specify cfs HBASE-14780 ADDENDEUM -
 remove extra imports from branch-1.0

Conflicts:
	hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestDDLMasterFailover.java (not in branch-1)
	hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestBulkLoad.java (imports fixes)

Change-Id: I19cfc02a5a1351d14cf320ba5d17b010b2e7f6e7
Reason: Testing
Author: Jonathan Hsieh
Ref: CDH-34128
---
 .../apache/hadoop/hbase/IntegrationTestBase.java   |    8 ++++++++
 .../hbase/IntegrationTestRegionReplicaPerf.java    |    4 +++-
 .../hbase/mapreduce/IntegrationTestBulkLoad.java   |    4 +++-
 .../IntegrationTestTableSnapshotInputFormat.java   |    8 +++-----
 4 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestBase.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestBase.java
index 097673a..3929524 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestBase.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestBase.java
@@ -203,7 +203,15 @@ public abstract class IntegrationTestBase extends AbstractHBaseTool {
 
   public abstract int runTestFromCommandLine() throws Exception;
 
+  /**
+   * Provides the name of the table that is protected from random Chaos monkey activity
+   * @return table to not delete.
+   */
   public abstract TableName getTablename();
 
+  /**
+   * Provides the name of the CFs that are protected from random Chaos monkey activity (alter)
+   * @return set of cf names to protect.
+   */
   protected abstract Set<String> getColumnFamilies();
 }
diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestRegionReplicaPerf.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestRegionReplicaPerf.java
index 3c29f4c..d05e039 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestRegionReplicaPerf.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestRegionReplicaPerf.java
@@ -19,6 +19,7 @@
 package org.apache.hadoop.hbase;
 
 import com.google.common.base.Objects;
+import com.google.common.collect.Sets;
 import com.yammer.metrics.core.Histogram;
 import org.apache.commons.cli.CommandLine;
 import org.apache.commons.logging.Log;
@@ -33,6 +34,7 @@ import org.apache.hadoop.hbase.client.Admin;
 import org.apache.hadoop.hbase.ipc.RpcClient;
 import org.apache.hadoop.hbase.regionserver.DisabledRegionSplitPolicy;
 import org.apache.hadoop.hbase.testclassification.IntegrationTests;
+import org.apache.hadoop.hbase.util.Bytes;
 import org.apache.hadoop.hbase.util.YammerHistogramUtils;
 import org.apache.hadoop.mapreduce.Counters;
 import org.apache.hadoop.mapreduce.Job;
@@ -234,7 +236,7 @@ public class IntegrationTestRegionReplicaPerf extends IntegrationTestBase {
 
   @Override
   protected Set<String> getColumnFamilies() {
-    return null;
+    return Sets.newHashSet(Bytes.toString(PerformanceEvaluation.FAMILY_NAME));
   }
 
   /** Compute the mean of the given {@code stat} from a timing results. */
diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestBulkLoad.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestBulkLoad.java
index 13da11a..a85bc9a 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestBulkLoad.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestBulkLoad.java
@@ -29,6 +29,7 @@ import java.util.Set;
 import java.util.concurrent.atomic.AtomicLong;
 
 import com.google.common.base.Joiner;
+import com.google.common.collect.Sets;
 import org.apache.commons.cli.CommandLine;
 import org.apache.commons.lang.RandomStringUtils;
 import org.apache.commons.logging.Log;
@@ -765,7 +766,8 @@ public class IntegrationTestBulkLoad extends IntegrationTestBase {
 
   @Override
   protected Set<String> getColumnFamilies() {
-    return null;
+    return Sets.newHashSet(Bytes.toString(CHAIN_FAM) , Bytes.toString(DATA_FAM),
+        Bytes.toString(SORT_FAM));
   }
 
   public static void main(String[] args) throws Exception {
diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestTableSnapshotInputFormat.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestTableSnapshotInputFormat.java
index 55da93e..1a152e8 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestTableSnapshotInputFormat.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/mapreduce/IntegrationTestTableSnapshotInputFormat.java
@@ -174,15 +174,13 @@ public class IntegrationTestTableSnapshotInputFormat extends IntegrationTestBase
     return 0;
   }
 
-  @Override // CM is not intended to be run with this test
+  @Override // Chaos Monkey is not intended to be run with this test
   public TableName getTablename() {
     return null;
   }
 
-  @Override
-  protected Set<String> getColumnFamilies() {
-    return null;
-  }
+  @Override // Chaos Monkey is not inteded to be run with this test
+  protected Set<String> getColumnFamilies() { return null; }
 
   public static void main(String[] args) throws Exception {
     Configuration conf = HBaseConfiguration.create();
-- 
1.7.9.5

